### PR TITLE
Notifying instead of failing fixes #2111

### LINKED ISSUE #2111: Don't fail because of key on table rename
https://rollbar.com/vizzuality/CartoDB/items/3963/occurrences/4104034917/

cc @iriberri

### PR DESCRIPTION
**Pull request by juanignaciosl**
@rambo what do you think about this? I think you told me Redis metadata (behind `$table_metadata`) is no longer used. There're situations like the reported bug when key (table name) is missing and that caused to  fail. I think recovering here is legit. I'm not confident about fully removing Redis table metadata usage yet, that's a whole different story now.

What to test: registration was previously done at Redis, and that's been removed. Everything should work fine. With an eye on Rollbar to avoid adding new errors.
* Importing.
* Manual new table.
* Geocoding.
* Adding and removing different types of features in a map, and rendering.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1448/))

**Comment by juanignaciosl** [comment]
@rambo @Kartones could you please CR this?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1466/))

**Comment by Kartones** [comment]
+1  and I wonder when we'll fully delete that  migrator20.rb and  migrator21.rb... probably with 3.0

**Comment by zenitraM** [comment]
The only user of this is/was the tiler, so probably @rochoa can confirm the Redis keys are not used anymore

**Comment by rochoa** [comment]
I'm actually removing old api (/tiles/:table/...), check: https://github.com/CartoDB/Windshaft-cartodb/issues/257

That is using the privacy metadata but since it was already deprecated I don't care too much about those endpoints failing randomly due to that change.

**Comment by juanignaciosl** [comment]
@rochoa : I think we should test this in staging with QA people then so nothing too harmful arises, could you tell me what should we test, related to the tiler? Thanks :-)

**Comment by rochoa** [comment]
If table privacy changes from "public" to "private" it will fail to render the tile either because the privacy is not enough (from the redis metadata) or because it has not enough database privileges.

I will double check the different paths and will write here what to test.

**Comment by juanignaciosl** [comment]
Ok, thank you very much. I'll wait for this to continue. I'm sorry for asking, but could you give an ETA? This Redis error is also raised at common data metadata loading. I was waiting to end this task to enable metadata loading, but if it's taking some days I might patch it instead.

**Comment by rochoa** [comment]
What's that "Redis error is also raised at common data metadata loading" error? I'm not aware of it.

ETA: I will do it tomorrow. :smile:

**Comment by juanignaciosl** [comment]
I'm speaking about #2134, I should've linked it before since I spoke about it, sorry :sweat_smile:
Tomorrow is great, thanks :-)

**Comment by javisantana** [comment]
hey, if the old api fails is not a problem

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1509/))

**Comment by juanignaciosl** [comment]
@rochoa : @Xatpy found a problem in staging (you can check now at ded01) that I've confirmed in local. This branch makes Varnish invalidation stop working, at least at one case. I've checked the changes and it doesn't seem to be a straightforward problem (I mean, I haven't deleted any call to Varnish invalidation), so you can probably help me or, at least, point to me the place where that cache should be invalidated.

Steps to reproduce:
* Open a map.
* Create a new polygon with "add new feature" button and "done" for finish.
* Expected: polygon is added an appears. What happens: polygon won't appear, but if you zoom, it will. It certainly looks like an invalidation issue (it didn't happen when I was not running Varnish locally).

Thank you in advance :)

File: app/models/table/privacy_manager.rb [hunk 1]
```diff
@@ -56,10 +56,9 @@ def privacy_for_redis
       end
     end
 
-    def propagate_to_redis_and_varnish
+    def propagate_to_varnish
       raise 'table privacy cannot be nil' unless privacy
       # TODO: Improve this, hack because tiler checks it
-      $tables_metadata.hset redis_key, 'privacy', privacy_for_redis
```
Comment (rochoa) [inline app/models/table/privacy_manager.rb]: BTW I would remove the `privacy_for_redis` method

**Comment by zenitraM** [comment]
@juanignaciosl let me look at the Varnish issue, it could have been my fault

**Comment by rochoa** [comment]
Things I've been testing:

Preconditions:
 - Have a table `wadus_table` with privacy set to `private`.
 - You are **NOT** able to retrieve http://{user}.cartodb.com/tiles/wadus_table/0/0/0.png

Test scenario:
1. Change privacy in `wadus_table` from `private` to `public`
1. You **CAN** retrieve http://{user}.cartodb.com/tiles/wadus_table/0/0/0.png
1. Change privacy in `wadus_table` from `public` back to `private`
1. You are **NOT** able to retrieve http://{user}.cartodb.com/tiles/wadus_table/0/0/0.png

Although we are not `hset`ing the privacy in Redis anymore privacy will continue working because at the end it relies postgres permissions to access the data.

HTH.

**Comment by juanignaciosl** [comment]
@rochoa do you mean that hset of the privacy made it work? :-\

@zenitraM I don't think it's your fault, because rui keeps working.

**Comment by rochoa** [comment]
What I mean is that the `hset` probably was a 'nice to have' thing at the time but it has no effect because it ends relying on database real permission for the user using the table.

**Comment by rochoa** [comment]
Have a look at this: https://github.com/CartoDB/cartodb/issues/1629

**Comment by juanignaciosl** [comment]
It is similar, but it won't appear, no matter how much I wait.

**Comment by rochoa** [comment]
If data changed the {layergroupid}:{last_updated_at} tuple should change so it should not hit anything cached.

Are you using private tables?

**Comment by juanignaciosl** [comment]
I created a new table at both with the defaults for my users, public in staging and private in local, but in local a public one also works. After drawing the polygon I see a difference within two POST requests. This request:
`https://juanignaciosl-ded.cartodb-staging.com/api/v1/map?map_key=48d11578f1fce30b3f3a59d131846a99a7f12e37`
returns 
```
cdn_url: {http: "api-staging-ded01.cartocdn.com", https: "cartocdn-staging-ded01.global.ssl.fastly.net"}
http: "api-staging-ded01.cartocdn.com"
https: "cartocdn-staging-ded01.global.ssl.fastly.net"
last_updated: "1970-01-01T00:00:00.000Z"
layergroupid: "f4c7eb1388d0fb7cb487c253787af386:0"
````
but the similar one in local (`http://development.localhost.lan/api/v1/map?map_key=cc654eec2c22b45b2a46e4938370530add62a2c5`) returns

```
last_updated: "2015-02-11T15:44:30.368Z"
layergroupid: "79ba7f70276ef1db79c2d9888a26fc1d:1423669470368.13"
```

In the staging request there's cdn information, and `last_updated` seems wrong. That request is the last one in staging, in local there're more afterwards that will refresh the content.

**Comment by juanignaciosl** [comment]
For future reference and the rest of you: @rochoa has been helping me with this. The problem was ded01 was using a Windshaft version from June. After upgrading it works as expected. Will continue testing then.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1529/))

**Comment by rochoa** [comment]
Yes, mystery solved! As @juanignaciosl said: Maps API in ded01 machine was running an old version with no x-cache-channel header in layergroups.

Thanks to @zenitraM all staging servers are being deployed, previously only api01-st was being deployed. So I hope no more issues like this inconsistency in staging environment!

**Comment by juanignaciosl** [comment]
Got +1 from QA, I'm merging & deploying.

**Pull request merged**